notebooks/NIRISS_WFSS_advanced/02_niriss_wfss_spec2.ipynb

```diff
@@ -0,0 +1,1155 @@
+   "id": "974ab7ce-d1dd-4bcc-b405-25c6dd683a25",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Update the CRDS path to your local directory\n",
+    "%env CRDS_PATH=crds_cache\n",
+    "%env CRDS_SERVER_URL=https://jwst-crds.stsci.edu"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9443f2ff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import glob\n",
+    "import json\n",
+    "import os\n",
+    "import shutil\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "from astropy.table import Table\n",
+    "from astropy.io import fits\n",
+    "\n",
+    "from jdaviz import Mosviz\n",
+    "from matplotlib import pyplot as plt\n",
+    "from matplotlib import patches\n",
+    "%matplotlib inline\n",
+    "\n",
+    "from jwst.pipeline import Spec2Pipeline\n",
+    "from jwst import datamodels"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "64b6e8d7-64a0-4a3b-9776-29b181452ff9",
+   "metadata": {},
+   "source": [
+    "Check what version of the JWST pipeline you are using. To see what the latest version of the pipeline is available, check <a href=\"https://github.com/spacetelescope/jwst#software-vs-dms-build-version-map\">here</a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "87cdec65",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import jwst\n",
+    "print('jwst',jwst.__version__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "436ed7e4-f642-41d0-83dd-565c24f78d5c",
+   "metadata": {},
+   "source": [
+    "#### Data Setup\n",
+    "\n",
+    "The data directory, `data_dir` should contain all of the association and rate files in a single, flat directory. `custom_run_image3` should match the output directory for the custom image3 calibration in the notebook 01_niriss_wfss_image2_image3.ipynb.\n",
+    "\n",
+    "For spec2, we need the rate files that we downloaded as well as the segmentation maps and source catalogs from image3. Because of that, we will create a new directory (defined by `custom_run_spec2`), change into it, and copy the relevant data over. \n",
+    "\n",
+    "*For a regular workflow, it is likely easier to leave all files in a single directory compared to the multiple directories we make here.*"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d218c8e0-7a25-4688-b904-c4225a66a8ca",
+   "metadata": {},
+   "source": [
+    "Open up our data information file which will make parsing the rate files to copy over easier as we only need the dispersed images, i.e. those observed with GR150R or GR150C."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "168c2146",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_dir = 'data'\n",
+    "custom_run_spec2 = 'custom_spec2'\n",
+    "custom_run_image3 = os.path.join(data_dir, 'custom_image3_calibrated') # results of custom image3 calibration\n",
```

Comment:
Similarly to notebook 01, you will have to download and expand the appropriate zip file:
https://data.science.stsci.edu/redirect/JWST/jwst-data_analysis_tools/niriss_wfss_advanced/niriss_wfss_advanced_02_input.zip